src/shogun/distance/EuclideanDistance.h

```diff
@@ -125,6 +161,21 @@ class CEuclideanDistance: public CRealDistance
 	protected:
 		/** if application of sqrt on matrix computation is disabled */
 		bool disable_sqrt;
+
+		/** if squared norms of rhs features are used */
+		bool m_rhs_sq_norms;
+
+		/** if squared norms of lhs features are used */
+		bool m_lhs_sq_norms;
+
+		/** i dot products are used in distance computations */
+		bool m_dot_enabled;
+
+		/** squared norms from features of right hand side to be used if dot enabled */
+		SGVector<float64_t> m_rhs_squared_norms;
```

Comment:
- Decided to make `precompute_xxx_squared_norms()` public since we don't know how often the rhs is changing.
for example: 
 - if rhs changes and we end up computing pairwise distances between say _all_ vectors of rhs and lhs this is worth.
 - But if rhs changes quite often and we  need pairwise distances using  only _some_  vectors , computing squared norms each time would be inefficient ?